include/affinity.unix.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <pthread.h>
+#include <sched.h>
+#include <unistd.h>
+
+#include "affinity.h"
+
+void affinity_int_base_mask() {
+	static bool initialized = false;
+	if(initialized) return;
+
+	assert(pthread_getaffinity_np(pthread_self(), sizeof(cpu_set_t), &g_affinity_base_mask) == 0 && "Error retrieving base affinity mask.");
+	initialized = true;
+}
+
+uint32 affinity_cores_available() {
+	affinity_int_base_mask();
```

Comment:
This is a very C-style implementation -- in C++ we have better methods for ensuring that the base affinity mask is initialized by the time it is accessed. (i.e. static initialization)